README.md



Note for Mac OS X: It seems cursor visibility settings are not supported, meaning that the cursor will always be visible.



CHANGELOG.md



   **Version 0.7.15 — Febuary 19, 2019**

 - Fixed potential compile errors due to relative paths in autoconf parameters.
 - Replaced obsolete AC\_TRY\_LINK with AC\_LINK\_IFELSE.
 - Updated tiny-xml-doc-tools.
 - Fixed underscores in markdown files.

---


   **Version 0.7.14 — September 3, 2017**

 - Renamed copyright files to “license” for github license detection compatibility, see [Github's “Licensing a repository”](https://help.github.com/articles/licensing-a-repository/) for further reference.
 - Added missing contributor phrasing to BSD-3 clause. The resulting license now exactly matches the wording used on Github and so also makes the license detection work.
 - Updated man page.

---


   **Version 0.7.13 — April 9, 2017**

 - Show error message in case no installed locales can be found.
 - Adapted to replacement of en\_US locale with en\_GB from libfizmo.

---


   **Version 0.7.12 — October 9, 2016**

 - Adapted to new split-library bulids.
 - Updated man page.

---


   **Version 0.7.11 — August 31, 2016**

 - Fixed typo in manpage.
 - Used tiny-xml-doc-tools for documentation.
 - Adapted to renaming “libcellif” to “libmonospaceif”.

---


   **Version 0.7.10 — July 28, 2016**

 - Updated manpage and coyright data for version 0.8.0 release.

---


   **Version 0.7.9 — October 9, 2015**

 - No more strict dependency on libncursesw since libncurses works equally well.
 - Added forgotten configuration options “with-ncurses-includedir”, “with-ncursesw-libdir” and “with-ncurses-libdir”.
 - Minor autoconf-related changes for fizmo-dist-0.7.10-compatibility.

---


   **Version 0.7.8 — March 19, 2014**

 - Adapted to automake v1.14 “subdir-objects” option.
 - Added missing license/copyright information.

---


   **Version 0.7.7 — June 5, 2013**

 - Adapted to new “fizmo\_register\_screen\_interface” and “fizmo\_start” functions.
 - Fix build error for some systems due to missing wchar.h include – thanks to Nikos Chantziaras.
 - Added default terminal color evluation.
 - Adapted to new “install-data-local” build target.

---


   **Version 0.7.6 — December 27, 2012**

 - Improved configure-detection for libncursesw for systems without pkg-config's ncursesw.pc.
 - Adapted to new “disable-x11” and “disable-sound” configuration.

---


   **Version 0.7.5 — November 30, 2012**

 - Fixed missing delete input.
 - Fixed manpage typos, thanks to Johan Ljunglid.
 - Implemented redraw via CTRL-R.

---


   **Version 0.7.4 — September 17, 2012**

 - Adapted to new autoconf/automake build system.

---


   **Version 0.7.3 — August 1, 2012**

 - Merged Andrew Plotkin's iOS-fizmo changes: Adapted to “prompt\_for\_filename” function in screen-cell-interface.
 - Minor memory leak cleanups.

---


   **Version 0.7.2 — March 9, 2012**

 - Fixed missing $(DESTDIR) variable – should already have been present since version 0.7.1.

---


   **Version 0.7.1 — November 6, 2011**

 - Fixed missing “override” statments for CFLAGS in Makefiles.
 - Fixed $(DESTDIR) evaluation in config.[default|macports].mk.
 - Changed manpage install dir from “man” to “share/man”.
 - Adapted Makefiles and configuration to use standard GNU Makefile variables instead of INSTALL\_PATH and FIZMO\_BIN\_DIR.
 - Respect LD\_FLAGS for linking.

---


   **Version 0.7.0 — September 18, 2011**

 - The “fizmo-ncursesw” interface has been re-implemented for version 0.7. It now makes use of libcellif for output. It uses libfizmo's new word hyphenation and implements better X11-output through libdrilbo.
 - X11 frontispiece display is now by default in a separate window, which, contrary to the older implementation that simply invoked a XCopyArea directly into the terminal's X window, should work reliably. For terminals which by chance support it, the old behaviour is still available by using the “display-x11-inline-image” config option. The inline display has been changed to no longer write directly into the terminal window, but instead to use a subwindow instead.
 - Adapted to version 0.7 locale system and the new build process.

---


   **Version 0.6.9 — April 24, 2011**

 - Added NCURSESW\_INCLUDE\_FROM\_W\_DIRNAME to configs to allow usage of Apple's ncurses implementation, added GDKPIXBUF\_INC\_DIR configuration variable.

---


   **Version 0.6.4 — May 24, 2009**

 - Bugfix release: Definitely recommend for read\_char fix.
 - Fixed interface to supply default colors in case ncurses' “pair\_content” returned invalid colors (thanks to Samuel Verschelde).
 - Input from read\_char is no longer re-echoed to the screen (thanks to Samnuel Verschelde).

---


   **Version 0.6.0 — March 25, 2009**

 - This marks the point for the first public beta release.
 - Implemented conversion of font 3 (charachter font) to unicode. This appears to make Beyond Zork's map appear at least readable.
 - Implemented recording and replaying of timed input.
 - The filename input for save and restore may now be cancelled using the escape key.
 - Pressing arrow up/down during filename input no longer displays the command history.
 - Implemented wordwrapper / ncursesw margins.
 - Implemented “no-update” command line flag and config option to avoid long startup scan times (for example on slow notebook drivers with many story files).
 - Implemented “--search” and “--recursively-search” command line invocations.
 - Fixed bug in scrollback parapgraph position cache.
 - Fixed scroll-down display for scrolling to bottom.
 - Fixed crash that occured when a read-instruction was recursively called from a timed input verification-routine.
 - Fixed display error on refresh of preloaded input from history.
 - Implemented forced predictable mode (ignores seeding random generator).
 - Many, many other minor fixes and extensions.

---


   **Version 0.5.3 — March 11, 2009**

 - Re-wrote Makefile system. Releveant module defaults are now kept in separate .mk-files.

src/locales/fizmo_ncursesw_locales.c

/* fizmo_ncursesw_locales.c
 *
 * This file is part of fizmo.
 *
 * Copyright (c) 2009-2017 Christoph Ender.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. Neither the name of the copyright holder nor the names of its
 *    contributors may be used to endorse or promote products derived
 *    from this software without specific prior written permission.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
 * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
 * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
 * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
 * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */


#ifndef fizmo_ncursesw_locales_c_INCLUDED
#define fizmo_ncursesw_locales_c_INCLUDED

#include <tools/types.h>

z_ucs fizmo_ncursesw_module_name[] = {
  'f', 'i', 'z', 'm', 'o', '_', 'n', 'c', 'u', 'r', 's', 'e', 's', 'w', 0 };

#endif // fizmo_ncursesw_locales_c_INCLUDED



src/locales/fizmo_ncursesw_locales.h

/* fizmo_ncursesw_locales.h
 *
 * This file is part of fizmo.
 *
 * Copyright (c) 2009-2017 Christoph Ender.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. Neither the name of the copyright holder nor the names of its
 *    contributors may be used to endorse or promote products derived
 *    from this software without specific prior written permission.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
 * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
 * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
 * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
 * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
 * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */


#ifndef fizmo_ncursesw_locales_h_INCLUDED
#define fizmo_ncursesw_locales_h_INCLUDED

#define i18n_ncursesw_SET_FOREGROUND_COLOR 0
#define i18n_ncursesw_SET_BACKGROUND_COLOR 1
#define i18n_ncursesw_SET_LEFT_MARGIN_SIZE 2
#define i18n_ncursesw_SET_RIGHT_MARGIN_SIZE 3
#define i18n_ncursesw_INVALID_CONFIGURATION_VALUE_P0S_FOR_P1S 4
#define i18n_ncursesw_USAGE_DESCRIPTION 5
#define i18n_ncursesw_VALID_OPTIONS_ARE 6
#define i18n_ncursesw_SET_LOCALE_NAME_FOR_INTERPRETER_MESSAGES 7
#define i18n_ncursesw_START_WITH_RANDOM_GENERATOR_IN_PREDICTABLE_MODE 8
#define i18n_ncursesw_START_GAME_WITH_TRANSCRIPT_ENABLED 9
#define i18n_ncursesw_LIBFIZMO_VERSION_P0S 10 // no longer used.
#define i18n_ncursesw_LOCALES_AVAILIABLE 11
#define i18n_ncursesw_MORE_PROMPT 12
#define i18n_ncursesw_SCORE 13
#define i18n_ncursesw_TURNS 14
#define i18n_ncursesw_NUMBER_OF_COLOR_COMBINATIONS_AVAILABLE_P0D 15
#define i18n_ncursesw_COLORS_AVAILABLE 16
#define i18n_ncursesw_INVALID_COLOR_NAME 17
#define i18n_ncursesw_SHOW_HELP_MESSAGE_AND_EXIT 18
#define i18n_ncursesw_USE_BOLD_FOR_BRIGHT_FOREGROUND_COLORS 19
#define i18n_ncursesw_USE_BLINK_FOR_BRIGHT_BACKGROUND_COLORS 20
#define i18n_ncursesw_DONT_USE_COLORS 21
#define i18n_ncursesw_ENABLE_COLORS 22
#define i18n_ncursesw_NO_STORIES_REGISTERED_PLUS_HELP 23
#define i18n_ncursesw_SPACE_FOR_NEXT_PAGE 24
#define i18n_ncursesw_B_FOR_LAST_PAGE 25
#define i18n_ncursesw_USE_UMEM_FOR_SAVEGAMES 26
#define i18n_ncursesw_DISABLE_X11_GRAPHICS 27
#define i18n_ncursesw_USE_XTERM_TITLE 28
#define i18n_ncursesw_FORCE_8BIT_SOUND 29
#define i18n_ncursesw_DISABLE_SOUND 30
#define i18n_ncursesw_SET_TANDY_FLAG 31
#define i18n_ncursesw_DONT_UPDATE_STORY_LIST_ON_START 32
#define i18n_ncursesw_SEARCH_DIRECTORY 33
#define i18n_ncursesw_RECURSIVELY_SEARCH_DIRECTORY 34
#define i18n_ncursesw_START_GAME_WITH_RECORDING_COMMANDS 35
#define i18n_ncursesw_START_GAME_WITH_INPUT_FROM_FILE 36
#define i18n_ncursesw_UPDATE_STORY_LIST_ON_START 37
#define i18n_ncursesw_START_WITH_RANDOM_GENERATOR_IN_RANDOM_MODE 38
#define i18n_ncursesw_SYNC_TRANSCRIPT 39
#define i18n_ncursesw_FUNCTION_CALL_P0S_ABORTED_DUE_TO_ERROR 40
#define i18n_ncursesw_FUNCTION_CALL_P0S_RETURNED_ERROR_P1D_P2S 41
#define i18n_ncursesw_ERROR_P0D_OCCURED_BEFORE_READ_P1S 42
#define i18n_ncursesw_UNKNOWN_ERROR_CASE 43
#define i18n_ncursesw_FUNCTION_CALL_MALLOC_P2D_RETURNED_NULL_PROBABLY_OUT_OF_MEMORY 44
#define i18n_ncursesw_FIZMO_NCURSESW_VERSION_P0S 45
#define i18n_ncursesw_LIBMONOSPACEINTERFACE_VERSION_P0S 46
#define i18n_ncursesw_LOCALE_SEARCH_PATH 47
#define i18n_ncursesw_FILENAME_TO_READ_COMMANDS_FROM 48
#define i18n_ncursesw_ENABLE_X11_INLINE_GRAPHICS 49
#define i18n_ncursesw_LIBDRILBO_VERSION_P0S 50
#define i18n_ncursesw_DISABLE_HYPHENATION 51
#define i18n_ncursesw_SET_TRANSCRIPT_FILENAME 52
#define i18n_ncursesw_FILENAME_TO_RECORD_INPUT_TO 53
#define i18n_ncursesw_COULD_NOT_OPEN_OR_FIND_P0S 54
#define i18n_ncursesw_SET_NUMBER_OF_MAXIMUM_UNDO_STEPS 55

extern z_ucs fizmo_ncursesw_module_name[];

#endif // fizmo_ncursesw_locales_h_INCLUDED

